### ISSUE TITLE
For gelu besed model, do we have same result?

### ISSUE THREAD

**Issue by jeremyyx** [thunlp/modularity-analysis#1]

**Comment by Zhiyuan-Zeng** [comment]
Hi @YixinSong-e, thanks for your interest in our work!

- For the neuron-level analysis and modularity analysis on pre-MoE Transformers, our framework is not designed specifically for ReLU-based Transformers. It can be utilized for modularity analysis on GeLU-based pre-MoE Transformers, and it is expected that similar results could be achieved with such analyses.

- In our limitations section, we acknowledged that one major limitation is the absence of research on GeLU-based vanilla Transformers. To elaborate, the MoEfication technique employed by our study isn’t applicable to GeLU-based Transformers. This technique is used to transform vanilla Transformers into their corresponding post-MoE versions. Therefore, future studies could focus on enhancing the MoEfication technique or investigate alternative technical approaches for studying modularity in GeLU-based vanilla Transformers.

**Comment by jeremyyx** [comment]
Thanks for reply! Actually, I have found that using a Gelu-based model and taking the top 60% of activation values can lead to some loss of accuracy. Maybe find a fast way to transform a Gelu besed model to a Relu based model is a solution.

**Comment by zzy14** [comment]
@YixinSong-e You're right. We also observe similar results when directly making a GeLU-based model sparse. This is likely because GeLU, unlike ReLU, does not zero out negative values but assigns them lower values, which still influence model performance. [Here](https://github.com/thunlp/MoEfication#tips-for-training-relu-based-bert) is an example to transform a GeLU-based model to a ReLU-based model, requiring only hundreds of training steps.

**Comment by jeremyyx** [comment]
Acutally I want to test that replace swiglu with reglu to test llama-6B model .I would like to confirm whether this fine-tuning can maintain accuracy for models with more parameters.  Do you think this is feasible?

**Comment by zzy14** [comment]
Yes, I think it is feasible. I also tried to replace swiglu with reglu for llama-7B. I found that fine-tuning on only 1B tokens of C4 leads to a loss of 2.1, close to the loss of the original model (about 2.0). For comparison, the pre-training corpus of the llama contains 1T tokens.

**Comment by jeremyyx** [comment]
Sounds good！ Can achieving the same performance as the original model be possible by adding more tokens? Are there any performance losses on other benchmarks?

**Comment by zzy14** [comment]
Yes, it's possible that you could achieve the same performance as the original model by adding more tokens to the fine-tuning process. Besides, it would be better to follow the data recipe of llama. Regarding your second question, it is a preliminary attempt and we did not evaluate the relu model on other benchmarks.

**Comment by YixinSong-e** [comment]
Can you share the weight file? I want to evaluate this model. I want to do some system research with sparse action characteristic. I am a Ph.D. student from Shanghai Jiao Tong University. May I have further communication with you?
